examples/metallic_systems.jl

```diff
@@ -0,0 +1,52 @@
+# # Temperature and metallic systems
+#
+# In this example we consider the modeling of a magnesium lattice
+# as a simple example for a metallic system.
+# First we import required packages and setup the lattice
+# using the magnesium GTH pseudopotential suitable for PBE,
+# the functional we are going to use.
+
+using DFTK
+using Plots
+
+a = 3.0179389193174084  # bohr
+b = 5.227223542397263   # bohr
+c = 9.773621942589742   # bohr
+lattice = [[-a -a  0]; [-b  b  0]; [0   0 -c]]
+Mg = ElementPsp(:Mg, psp=load_psp("hgh/pbe/Mg-q2"))
+atoms = [Mg => [[2/3, 1/3, 1/4], [1/3, 2/3, 3/4]]];
+
+# Next we build the PBE model and discretize it.
+# Since magnesium is a metal we apply a small smearing
+# temperature to ease convergence using the Methfessel-Paxton
+# smearing scheme. Note that both the `Ecut` is too small
+# as well as the minimal ``k``-point spacing
+# `kspacing` far too large to give a converged result.
+# The goal of them was to obtain a fast execution time.
+
+kspacing = 0.5     # Minimal spacing of k-points
+Ecut = 5           # kinetic energy cutoff in Hartree
+Tsmear = 0.01      # Smearing temperature in Hartree
+
+kgrid = kgrid_size_from_minimal_spacing(lattice, kspacing)
+kgrid_size_from_minimal_spacing(lattice, kspacing)
+model = model_DFT(lattice, atoms, [:gga_x_pbe, :gga_c_pbe];
+                  temperature=Tsmear,
+                  smearing=DFTK.Smearing.MethfesselPaxton1())
+basis = PlaneWaveBasis(model, Ecut, kgrid=kgrid);
+
+# Finally we run the SCF solving for 8 bands and print
```

Comment:
Explain a bit that we need more bands to ensure we need to capture partially occupied bands